### ISSUE TITLE
[Bug] Insertion Query With Subquery For Value Uses 'DEFAULT' Value

### ISSUE THREAD

**Issue by ryanqc** [typeorm/typeorm#1178]
Driver: Postgres
Version: ^0.1.2

In version 0.1.0-alpha.50 the following worked fine (I didn't verify in the intermediate versions):

Given entities User and Bar where Bar is a OneToOne on user.

```
this.createQueryBuilder('b')
  .insert()
  .values({
    user: () => '(SELECT "user".id FROM "user" WHERE "user”.foo = :myFoo)',
    bar,
  })
  .setParameter(‘myFoo’,  someFooValue)
  .returning('*')
  .execute();
```

Results in Query:
```
INSERT INTO “bar”(“bar”, "user") VALUES ($1,DEFAULT) RETURNING *
```

My current work around is querying the User repository manually to grab the user id, which results in another unnecessary query.

The generated query in 0.1.0-alpha.50 worked great as expected per the `QueryPartialEntity<T>` type definition.

**Comment by pleerock** [comment]
Can you please check it once again in the latest version? Because I just added test and it passes fine

**Comment by pleerock** [comment]
```
executing query:  INSERT INTO "post"("name", "userId") VALUES ($1,(SELECT "user"."id" FROM "user" WHERE "user"."name" = $2)) RETURNING * -- PARAMETERS: ["First post","Timber Saw"]
executing query:  SELECT "Post"."id" AS "Post_id", "Post"."name" AS "Post_name", "Post"."userId" AS "Post_userId" FROM "post" "Post" WHERE "Post"."id"=$1 -- PARAMETERS: [1]

Post { id: 1, name: 'First post', userId: 1 }
```

**Comment by pleerock** [comment]
Okay so its not working in the case if `user` is a relation. Its hard to fix this issue in 0.1.x versions, so I made a fix in the next 0.2.x version, you can install it via `npm i typeorm@next`

**Comment by stevefan1999-personal** [comment]
I observed this issue with MySQL driver. Strangely enough, Postgres and SQLite are not affected
